app/scripts/models/account.js

```diff
@@ -350,15 +350,24 @@ define([
           .then(function (client) {
             profileClient = client;
             var accessToken = self.get('accessToken');
-            return profileClient[method].apply(profileClient, [accessToken].concat(args));
+            if (accessToken) {
+              return profileClient[method].apply(profileClient, [accessToken].concat(args));
+            } else {
+              throw ProfileClient.Errors.toError('UNAUTHORIZED');
```

Comment:
Hmm, the token fetch promise should not resolve if fetching the token fails... Will look more later.